service-account/bound-sa-token.md

```diff
@@ -1,122 +0,0 @@
-%title: Bound Service Account Token
-%author: Krzysztof Ostrowski
-%date: 10.03.2023
-
--> # Bound Service Account Token <-
-
---------------------------------------------------
--> # Table of Contents <-
-==============
-
--> # What is a Service Account Token  <-
--> # Service Account Token as a YAML <-
--> # Old Service Account Token Behavior  <-
--> # What is a Bound Service Account Token  <-
--> # Lord of Tokens: Kubelet <-
--> # Lord of ServiceAccounts: Service Account Admission Controller <-
--> # Bound Service Account Token in use by a Pod <-
-
--------------------------------------------------
--> # What is a Service Account Token  <-
-
-An **identity** that an instance or an application can use to make **API requests**:
-
-1. Intended to be used for **processes**.
-2. Bound to a **namespace**.
-3. Being associated with a ServiceAccount provides a **Bearer Token**.
-
--------------------------------------------------
--> # Service Account Token as a YAML <-
-
-```yaml
-apiVersion: v1
-kind: ServiceAccount
-metadata:
-  name: my-sa
-  namespace: my-namespace
-```
-
--------------------------------------------------
--> # Old Service Account Token Behavior  <-
-
-
-1. Previously **didn't expire**.
-2. Previously could be **used anywhere**.
-3. Token was **stored in a secret**.
-
--------------------------------------------------
--> # What is a Bound Service Account Token  <-
-
-- A Service Account Token that is **bound** to an **audience and lifespan**.
-- Given by **volume projection** or **TokenRequestAPI**.
-
--------------------------------------------------
--> # Lord of Tokens: Kubelet <-
-
-- Tries to **rotate** it after 80% of its lifetime is over.
-- Acquires it from the `apiserver` through the `TokenRequestAPI`.
-
--------------------------------------------------
--> # Lord of ServiceAccounts: Service Account Admission Controller <-
-
-In case that a **Pod Manifest** doesn't have a `ServiceAccount` specified:
-- adds the default `ServiceAccount`,
-- adds a **projected volume** and
-- adds `ImagePullSecrets`
```

Comment:
not sure if `ImagePullSecrets` are important here.

However, it brings questions that you can expect from the audience:
- what's an ImagePullSecret? How is it represented? Where do I get it?